### ISSUE TITLE
[cmd/opampsupervisor] RemoteConfigStatus is not populated with failed on invalid config

### ISSUE THREAD

**Issue by Asarew** [open-telemetry/opentelemetry-collector-contrib#34785]
### Component(s)

cmd/opampsupervisor

### Is your feature request related to a problem? Please describe.

When passing down "invalid" configuration from the otel controller to the supervisor, the supervisor doesn't report back in the RemoteConfigStatus status == failed. It does report back Unhealthy with a LastError, but relying on that seems to break the opamp specification.

### Describe the solution you'd like

Call the collector validate command before starting and the agent. if that fails report the error message back in the RemoteConfigStatus with the correct status of Failed.

### Describe alternatives you've considered

Reading the ComponentHealth as the RemoteConfigStatus for now, but in my opinion that's a bad implementation of the opamp spec from both the controller as the supervisor.

**Comment by BinaryFissionGames** [comment]
Yep, this is absolutely something that's missing right now. It's tracked here:
https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/21079

Looks like there was a PR opened for this but it slipped through the cracks somehow.

**Comment by Asarew** [comment]
not stale :(
/label never-stale

**Comment by atoulme** [comment]
@Asarew I'm not well placed to help here as I am not familiar with the spec enough. Do you have a link to the section of the spec this behavior breaks? Additionally, would it make sense to have a spec change to add the behavior you are asking for, or is it already specified there?

Lastly, would you be open to offer an integration test based on what you offered? That'd help heaps move this forward.
